File: Ejercicio 3/Ejer3.java
import java.util.Scanner;

class Avion {
    String nombre;
    float velocidad;
    float combustible;
    String salud;

    public Avion(String nombre, float velocidad, float combustible, String salud) {
        this.nombre = nombre;
        this.velocidad = velocidad;
        this.combustible = combustible;
        this.salud = salud;
    }

    public void despegar() {
        System.out.println("Despegando...");
    }

    public void aterrizar() {
        System.out.println("Aterrizando...");
    }

    public void volar() {
        System.out.println("Volando...");
    }
}

class Mision {
    String destino;
    String duracion;
    int tripulacion;

    public Mision(String destino, String duracion, int tripulacion) {
        this.destino = destino;
        this.duracion = duracion;
        this.tripulacion = tripulacion;
    }

    public void iniciar() {
        System.out.println("Iniciando mision...");
    }

    public void finalizar() {
        System.out.println("Finalizando mision...");
    }

    public void abortar() {
        System.out.println("Abortando mision...");
    }
}

public class Main {
    static Scanner scanner = new Scanner(System.in);

    public static int menuJuego() {
        System.out.println("1. Iniciar mision");
        System.out.println("2. Finalizar mision");
        System.out.println("3. Abortar mision");
        System.out.println("4. Despegar");
        System.out.println("5. Aterrizar");
        System.out.println("6. Volar");
        System.out.println("7. Verificar combustible");
        System.out.println("8. Verificar tripulacion");
        System.out.println("9. Salir");
        System.out.print("Ingrese una opción: ");
        return scanner.nextInt();
    }

    public static void main(String[] args) {
        int opcion, tripulacion, tiempoVolado = 0;
        float velocidad, combustible;
        String salud, destino, duracion, nombre;
        System.out.println("RECOMENDACIONES:");
        System.out.println("- El combustible debe ser mayor a 20");
        System.out.println("- La tripulacion debe ser mayor a 3");
        System.out.println("- La mision se completara al volar 100 minutos");
        System.out.println("- Si el combustible es menor a 10, fallaste en tu mision");
        System.out.println("- Si el combustible es menor a 20, se recomienda abortar mision");
        System.out.println("- Si la tripulacion es menor a 3, se recomienda abortar mision");
        System.out.println("- Si la mision se aborta, fallaste en tu mision");
        System.out.println();
        System.out.print("Ingrese el nombre del avion: ");
        nombre = scanner.next();
        System.out.print("Ingrese la velocidad del avion: ");
        velocidad = scanner.nextFloat();
        System.out.print("Ingrese el combustible del avion: ");
        combustible = scanner.nextFloat();
        System.out.print("Ingrese la salud del avion: ");
        salud = scanner.next();
        Avion avion = new Avion(nombre, velocidad, combustible, salud);
        System.out.println();
        System.out.print("Ingrese el destino de la mision: ");
        destino = scanner.next();
        System.out.print("Ingrese la duracion de la mision: ");
        duracion = scanner.next();
        System.out.print("Ingrese la tripulacion de la mision: ");
        tripulacion = scanner.nextInt();
        Mision mision = new Mision(destino, duracion, tripulacion);
        System.out.println();
        while (true) {
            opcion = menuJuego();
            if (avion.combustible < 10) {
                System.out.println("El combustible es muy bajo, fallaste en tu mision.");
                return;
            }
            switch (opcion) {
                case 1:
                    avion.despegar();
                    avion.combustible -= 10;
                    break;
                case 2:
                    avion.aterrizar();
                    break;
                case 3:
                    avion.volar();
                    tiempoVolado += 10;
                    avion.combustible -= 5;
                    if (tiempoVolado == 100) {
                        System.out.println("Mision completada!");
                        return;
                    }
                    break;
                case 4:
                    mision.iniciar();
                    break;
                case 5:
                    mision.finalizar();
                    break;
                case 6:
                    mision.abortar();
                    System.out.println("Mision fallida!");
                    return;
                case 7:
                    System.out.println("El combustible es de: " + avion.combustible);
                    if (avion.combustible < 20) {
                        System.out.println("El combustible es menor a 20, se recomienda abortar mision");
                    }
                    break;
                case 8:
                    System.out.println("La tripulacion es de: " + mision.tripulacion);
                    if (mision.tripulacion < 3) {
                        System.out.println("La tripulacion es menor a 3, se recomienda abortar mision");
                    }
                    break;
                case 9:
                    return;
                default:
                    System.out.println("Opcion invalida");
            }
        }
    }
}

File: Ejercicio 3/Ejer3.cpp
#include <iostream>
#include <vector>
using namespace std;

class Avion{
    
    public:
        string nombre;
        float velocidad;
        float combustible;
        string salud;

        Avion(string nombre, float velocidad, float combustible, string salud){
            this->nombre = nombre;
            this->velocidad = velocidad;
            this->combustible = combustible;
            this->salud = salud;
        }

        void despegar(){
            cout<<"Despegando..."<<endl;
        }

        void aterrizar(){
            cout<<"Aterrizando..."<<endl;
        }

        void volar(){
            cout<<"Volando..."<<endl;
        }
};

class Mision{

    public:
        string destino;
        string duracion;
        int tripulacion;

        Mision(string destino, string duracion, int tripulacion){
            this->destino = destino;
            this->duracion = duracion;
            this->tripulacion = tripulacion;
        }

        void iniciar(){
            cout<<"Iniciando mision..."<<endl;
        }

        void finalizar(){
            cout<<"Finalizando mision..."<<endl;
        }

        void abortar(){
            cout<<"Abortando mision..."<<endl;
        }
};
int menu_juego(){
    int opcion;
    cout<<"1. Iniciar mision"<<endl;
    cout<<"2. Finalizar mision"<<endl;
    cout<<"3. Abortar mision"<<endl;
    cout<<"4. Despegar"<<endl;
    cout<<"5. Aterrizar"<<endl;
    cout<<"6. Volar"<<endl;
    cout<<"7. Verificar combustible"<<endl;
    cout<<"8. Verificar tripulacion"<<endl;
    cout<<"9. Salir"<<endl;
    cout<<"Ingrese una opcion: ";
    cin>>opcion;
    return opcion;
}
int main(){
    int opcion, tripulacion, tiempo_volado;
    float velocidad, combustible;
    string salud, destino, duracion, nombre;
    cout<<"RECOMENDACIONES: "<<endl;
    cout<<"- El combustible debe ser mayor a 20"<<endl;
    cout<<"- La tripulacion debe ser mayor a 3"<<endl;
    cout<<"- La mision se completara al volar 100 minutos"<<endl;
    cout<<"- Si el combustible es menor a 10, fallaste en tu mision"<<endl;
    cout<<"- Si el combustible es menor a 20, se recomienda abortar mision"<<endl;
    cout<<"- Si la tripulacion es menor a 3, se recomienda abortar mision"<<endl;
    cout<<"- Si la mision se aborta, fallaste en tu mision"<<endl;
    cout<<endl;
    cout<<"Ingrese el nombre del avion: ";
    cin>>nombre;
    cout<<"Ingrese la velocidad del avion: ";
    cin>>velocidad;
    cout<<"Ingrese el combustible del avion: ";
    cin>>combustible;
    cout<<"Ingrese la salud del avion: ";
    cin>>salud;
    Avion avion(nombre, velocidad, combustible, salud);
    cout<<endl;
    cout<<"Ingrese el destino de la mision: ";
    cin>>destino;
    cout<<"Ingrese la duracion de la mision: ";
    cin>>duracion;
    cout<<"Ingrese la tripulacion de la mision: ";
    cin>>tripulacion;
    Mision mision(destino, duracion, tripulacion);
    cout<<endl;
    while(true){
        opcion = menu_juego();
        if(avion.combustible < 10){
            cout<<"El combustible es muy bajo, fallaste en tu mision."<<endl;
            return 0;
        }
        if(opcion == 1){
            avion.despegar();
            avion.combustible -= 10;
        } else if(opcion == 2){
            avion.aterrizar();
        } else if(opcion == 3){
            avion.volar();
            tiempo_volado += 10;
            avion.combustible -= 5;
            if(tiempo_volado == 100){
                cout<<"Mision completada!"<<endl;
                return 0;
            } 
        } else if(opcion == 4){
            mision.iniciar();
        } else if(opcion == 5){
            mision.finalizar();
        } else if(opcion == 6){
            mision.abortar();
            cout<<"Mision fallida!"<<endl;
            return 0;
        } else if(opcion == 7){
            cout<<"El combustible es de: "<<avion.combustible<<endl;
            if(avion.combustible < 20){
                cout<<"El combustible es menor a 20, se recomienda abortar mision"<<endl;
            }
        } else if(opcion == 8){
            cout<<"La tripulacion es de: "<<mision.tripulacion<<endl;
            if(mision.tripulacion < 3){
                cout<<"La tripulacion es menor a 3, se recomienda abortar mision"<<endl;
            }
        } else if(opcion == 9){
            return 0;
        }
    }
}